src/customGraphQLValidationRules.js

```diff
@@ -111,22 +113,26 @@ export function RequiredFields(context, options) {
     // enough to have some child fragment to include the field, since we don't
     // know if that fragment covers all of the possible type options.
     Field(node) {
-      const def = context.getFieldDef();
-      if (!def) {
+      const type = context.getType();
+
+      if(mustHaveSelection(type) && !node.selectionSet) {
+        context.reportError(
+          new GraphQLError(
+            `'${node.name.value}' field must have a selection since type '${type}' is a Composite Type`,
+            [node]
+          )
+        );
         return;
       }
 
       requiredFields.forEach(field => {
-        if (fieldAvailableOnType(def.type, field)) {
-          const fieldWasRequested = getFieldWasRequestedOnNode(node, field);
-          if (!fieldWasRequested) {
-            context.reportError(
-              new GraphQLError(
-                `'${field}' field required on '${node.name.value}'`,
-                [node]
-              )
-            );
-          }
+        if (fieldAvailableOnType(type, field) && !fieldIsRequestedOnNode(node, field)) {
+          context.reportError(
+            new GraphQLError(
+              `'${field}' field required on '${node.name.value}'`,
+              [node]
+            )
+          );
```

Comment:
I'm making a little refactor here, say it if you don't want it. I can revert and only keep the bug fix.